File: server-smc-dart/lib/core/middlewares/cors/cors_middleware.dart
import 'dart:io';

import 'package:server_smc/core/middlewares/middlewares.dart';
import 'package:shelf/shelf.dart';
import 'package:dotenv/dotenv.dart';

class CorsMiddleware extends Middewares {
  final Map<String, String> headers = {
    'Access-Control-Allow-Origin': '${getOrigins()}',
    'Access-Control-Allow-Methods': 'GET, POST, PUT, PATCH, DELETE, OPTIONS',
    'Access-Control-Allow-Headers':
        '${HttpHeaders.contentTypeHeader}, ${HttpHeaders.authorizationHeader}'
  };

  @override
  Future<Response> execute(Request request) async {
    if (request.method == 'OPTIONS') {
      return Response(HttpStatus.ok, headers: headers);
    }
    final response = await innerHandler(request);

    return response.change(headers: headers);
  }
}

getOrigins() {
  return env['urlFrontEnd']!;
}

File: backend-smc-laravel/app/Http/Middleware/CorsMiddleware.php
<?php

namespace App\Http\Middleware;

use Closure;

class CorsMiddleware
{

    public function handle($request, Closure $next)
    {
        if($request->is('api/*')){
            header("Access-Control-Allow-Origin: *");
            header('Access-Control-Allow-Headers: Content-Type,Authorization');
            header('Access-Control-Allow-Methods: GET, POST, PUT, PATCH, DELETE');
            header('Access-Control-Expose-Headers: Authorization');
        }

        return $next($request);
    }
}
